ocpkg

```diff
@@ -867,6 +868,12 @@ install_opencog_github_repo cogserver
 # Install OpenCog
 install_opencog(){
 install_opencog_github_repo opencog
+install_opencog_github_repo miner
+install_opencog_github_repo learn
+install_opencog_github_repo attentionbank
+install_opencog_github_repo spacetime
+install_opencog_github_repo pattern-index
+install_opencog_github_repo visualization
 }
```

Comment:
Some of these are not dependencies.  So `pattern-index` is abandon-ware; I think it is the result of a mis-understanding of how things work. Nothing depends on `learn` or on `visualization`. In principle, someday, maybe, ghost might want to depend on `spacetime`, but currently, it does not.  I don't think anything depends on `miner`, either.   The lack of dependencies is one reason I split them out.

Things do depend on attentionbank, but I think that better designs for it are possible, so the reason to split it out is to allow other, competing implementations to be created.  (I realized I needed something like attentinobank for learning, but the current implementation seemed inadequate/incomplete)

Reply:
I mean, its plausible to have an "install-everything" target, I suppose ... and it's plausible to make that target be `opencog` ... in which case, we should really install everything.  But in the meanwhile, that makes it hard for anyone who wants only ghost, and not other parts. Also, installing abaondonware is probably a bad idea.

Reply:
Yes, I understand that all of them except `ure` doesn't have dependent projects and they are leafs of dependency tree. My point is that people who use `octool` or `opencog-dev:cli` docker to install opencog and play with it will not have this components installed. Agree that adding `install-everything` option makes sense. But two considerations here: first it will require change in docker, and probably some documentation; second my understanding @AmeBel has idea of more elegant `octool` replacement, so I don't know if it makes sense to change something here. These components were installed before after all.

Reply:
Actually we are experimenting with `pattern-index`, so I would not say it is abandonware. Next step I was going to suggest is adding components listed in https://github.com/opencog/opencog/issues/3391#issuecomment-528239862 into CircleCI config because they contain unit tests which are not executed since they are moved out of `opencog`.